h/views.py

```diff
@@ -111,9 +113,63 @@ def stream(context, request):
         location = request.resource_url(context, 'stream', query=query)
         return httpexceptions.HTTPFound(location=location)
     else:
+        context["link_tags"] = [{
+            "rel": "alternate", "href": request.route_url("atom_stream"),
+            "type": "application/atom+xml"}]
         return context
 
 
+def _atom_stream_validate_limit(request):
+    """Return the limit (max number of annotations) to use for the Atom stream.
+
+    Return the ?limit=n param from the URL.
+    If that's is missing or invalid return the h.stream.atom.limit setting
+    from the config file.
+    If that's also missing or invalid return 10.
+
+    """
+    def default():
+        return request.registry.settings.get("h.stream.atom.limit")
```

Comment:
I think I'd prefer you to use `10` as the default in this `get` call. I think you've decided not to so that it can be the default whenever the setting doesn't validate, but I would rather not silently swallow configuration errors like this.

The way I would prefer to handle this would be to, just like the api base url, validate in the includeme function. It's a common Pyramid pattern to parse, normalize, validate etc config settings in these functions during application startup. I would raise `RuntimeError` in the includeme so we never have to deal with this once the application is running.